### PR TITLE
Merge to master

### PR DESCRIPTION
**Pull request by slackve**

### CODE REVIEW

File: sdk/websocket.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ var (
 	confirmedAddedChannels     = make(map[string]chan Transaction)
 )
 
+type subscribeInfo struct {
```
Comment (ruslanBik4) [inline sdk/websocket.go]: PLease, move private types definitions after public

File: sdk/websocket.go [hunk 2]
```diff
@@ -200,8 +204,12 @@ func (c *ClientWebsocket) subsChannel(s *subscribe) error {
 			}
 
 			subName, _ := restParser(resp)
+			b := subscribeInfo{
+				name:    subName,
+				account: s.getAdd(),
+			}
 
-			e = s.buildType(subName, resp)
+			e = b.buildType(resp)
```
Comment (ruslanBik4) [inline sdk/websocket.go]: The func working in goroutine. Do you sure that finish & return 'e' which is defined by this string?

File: sdk/websocket.go (hunk 1 above)
Comment (slackve) [reply sdk/websocket.go]: solved in the websocket branch

File: sdk/websocket.go (hunk 2 above)
Comment (slackve) [reply sdk/websocket.go]: solved in: https://github.com/proximax-storage/nem2-sdk-go/pull/115/commits/aa550fdc2f6d1a028381e55fef54bcdb79b2a107

File: sdk/network.go [hunk 3]
```diff
@@ -19,18 +20,19 @@ type networkDTO struct {
...
 	if err != nil {
 		return 0, resp, err
 	}
 
-	networkType := NetworkTypeFromString(netDTO.Name)
+	networkType = NetworkTypeFromString(netDTO.Name)
+
 	if networkType == NotSupportedNet {
-		err = errors.New("network " + netDTO.Name + " is not supported yet by the sdk")
+		err = errors.New(fmt.Sprintf("network %s is not supported yet by the sdk", netDTO.Name))
```
Comment (ruslanBik4) [inline sdk/network.go]: use fmt.Errorf

File: sdk/subscribe_block.go [hunk 4]
```diff
@@ -24,6 +25,96 @@ const (
...
+		delete(signerInfoChannels, s.getAdd())
+		close(chType)
+
+	case chan *ErrorInfo:
+		chType := s.Ch.(chan *ErrorInfo)
+		delete(errChannels, s.getAdd())
+		close(chType)
+
+	default:
+		chType := s.Ch.(chan Transaction)
```
Comment (ruslanBik4) [inline sdk/subscribe_block.go]: Type 'chan Transaction' checking should be done. We can get a panic when changing the code here.
default-case need return error 'WRONG TYPE CHANNEL'.
Also we need add case 'nil'.

File: sdk/subscribe_block.go [hunk 5]
```diff
@@ -24,6 +25,96 @@ const (
 	pathCosignature        = "cosignature"
 )
 
+// Closes the subscription channel.
+func closeChannel(s *subscribe) {
```
Comment (ruslanBik4) [inline sdk/subscribe_block.go]: I think you need to pass as a parameter channel immediately.
Or make method closeChannel as inside method *subcsribe

**Comment by alvin-reyes** [comment]
Merging changes to master.

**Pull request merged**